### PR TITLE
CI: use conda incubator in benchmarks

### PR DESCRIPTION
**Pull request by fangchenli**
The benchmark ci currently has the longest runtime. Using conda incubator would save about 6-8 min.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -91,26 +100,45 @@ jobs:
...
-        source activate pandas-dev
-        python web/pandas_web.py web/pandas --target-path=web/build
+        conda info
+        conda list
 
-    - name: Build documentation
+    - name: Build Pandas
       run: |
-        source activate pandas-dev
-        doc/make.py --warnings-are-errors | tee sphinx.log ; exit ${PIPESTATUS[0]}
```
Comment (fangchenli) [inline .github/workflows/ci.yml:113]: ${PIPESTATUS[0]} is 1 now. not sure why

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -91,26 +100,45 @@ jobs:
...
+        python setup.py build_ext -j 2
+        python -m pip install -e . --no-build-isolation --no-use-pep517
+
+    - name: Build website
+      run: python web/pandas_web.py web/pandas --target-path=web/build
+
+    - name: Build documentation
+      run: doc/make.py --warnings-are-errors | tee sphinx.log ; exit ${PIPESTATUS[0]}
```
Comment (fangchenli) [inline .github/workflows/ci.yml:141]: ${PIPESTATUS[0]} is 1 now, not sure why.

File: doc/source/user_guide/enhancingperf.rst [hunk 3]
```diff
@@ -247,7 +247,7 @@ We've gotten another big improvement. Let's check again where the time is spent:
 
 .. ipython:: python
 
-   %%prun -l 4 apply_integrate_f(df["a"].to_numpy(), df["b"].to_numpy(), df["N"].to_numpy())
```
Comment (fangchenli) [inline doc/source/user_guide/enhancingperf.rst:250]: `UsageError: %%prun is a cell magic, but the cell body is empty. Did you mean the line magic %prun (single %)?`

**Comment by jorisvandenbossche** [comment]
@fangchenli this looks like a nice clean-up in any case (thanks!), but how would it save 6-8 minutes? (it currently does not seem to take less time as on master, unless the caching is not yet working here in this PR?)

File: doc/source/user_guide/enhancingperf.rst (hunk 3 above)
Comment (jorisvandenbossche) [reply doc/source/user_guide/enhancingperf.rst:250]: Yes, this is a good catch, thanks

**Comment by fangchenli** [comment]
> @fangchenli this looks like a nice clean-up in any case (thanks!), but how would it save 6-8 minutes? (it currently does not seem to take less time as on master, unless the caching is not yet working here in this PR?)

My mistake... It's more like 3 minutes. The caching is not working in this PR (maybe it doesn't work on PRs?). The two database builds are using conda incubator currently. It takes less time to setup the environment on those two builds. So I think caching is working there.

**Comment by jreback** [comment]
caching is only enabled in the master branch pushes

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -15,61 +15,70 @@ jobs:
...
       run: ci/code_checks.sh patterns
       if: always()
 
-    - name: Setup environment and build pandas
-      run: ci/setup_env.sh
-      if: always()
+    - name: Cache conda
+      uses: actions/cache@v1
+      env:
+        CACHE_NUMBER: 0
```
Comment (jreback) [inline .github/workflows/ci.yml:33]: what does this do?

Comment (jreback) [reply .github/workflows/ci.yml:33]: just fix the hash key?

**Review by jreback — CHANGES_REQUESTED** [review]
weirdly the doc build is failing

File: .github/workflows/ci.yml (hunk 4 above)
Comment (fangchenli) [reply .github/workflows/ci.yml:33]: I copied from conda incubator. But I guess we don't need this.

**Comment by fangchenli** [comment]
> weirdly the doc build is failing

`doc/make.py --warnings-are-errors | tee sphinx.log ; exit ${PIPESTATUS[0]}`

`PIPESTATUS[0]` is 1 now. Not sure why. Looking at the output of `doc/make.py --warnings-are-errors`, nothing is failing.

**Comment by jreback** [comment]
maybe the doc build is not picking up the correct file / deps as it is in master?

**Comment by fangchenli** [comment]
> maybe the doc build is not picking up the correct file / deps as it is in master?

I fixed the error mentioned above. Right now the outputs of `doc/make.py --warnings-are-errors` of this PR is identical to master's. Only the exit codes are different.

**Comment by fangchenli** [comment]
I'll revert the doc build and open an issue for it.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @fangchenli

File: doc/source/whatsnew/v0.8.0.rst [hunk 5]
```diff
@@ -176,7 +176,7 @@ New plotting methods
 Vytautas Jancauskas, the 2012 GSOC participant, has added many new plot
 types. For example, ``'kde'`` is a new option:
 
-.. ipython:: python
+.. code-block:: python
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.8.0.rst:179]: @fangchenli do you remember why you needed to make this change?

Comment (fangchenli) [reply doc/source/whatsnew/v0.8.0.rst:179]: It caused an error while using incubator. But I didn't get the incubator to work for the doc build. I probably should revert it as well. 

```bash
>>>-------------------------------------------------------------------------
Exception in /home/runner/work/pandas/pandas/doc/source/whatsnew/v0.8.0.rst at block ending on line 201
Specify :okexcept: as an option in the ipython:: block to suppress this message
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-2-65b822729fc4> in <module>
----> 1 plt.figure()

NameError: name 'plt' is not defined
<<<-------------------------------------------------------------------------
```

**Pull request merged**